### PR TITLE
Expand documentation on concatenation operator safety

### PR DESCRIPTION
**Pull request by simshaun**
Helps clarify issue #2728

### CODE REVIEW

**Review by SpacePossum — APPROVED** [review]

File: doc/templates.rst [hunk 1]
```diff
@@ -782,9 +782,13 @@ The following operators don't fit into any of the other categories:
 
       (1..5)|join(', ')
 
-* ``~``: Converts all operands into strings and concatenates them. ``{{ "Hello
-  " ~ name ~ "!" }}`` would return (assuming ``name`` is ``'John'``) ``Hello
-  John!``.
+* ``~``: Converts all operands into strings and safely concatenates them (to
+  avoid potential XSS). ``{{ "Hello " ~ name ~ "!" }}`` would return
```
Comment (stof) [inline doc/templates.rst]: the operator is not about *safely* concatenating. It does nothing regarding safety (and so passing it to something assuming it is already safe from XSS would actually creates XSS). The auto-escaping is doing things for safety (what is important to understand is that auto-escaping does not get applied on variables, but on what gets output using `{{ ... }}`, which is why things get escape). the safety does not come from `~`, but from `{{ ... }}` (and the fact that auto-escaping is ON by default)

Comment (simshaun) [reply doc/templates.rst]: oooh

File: doc/templates.rst [hunk 2]
```diff
@@ -784,7 +784,10 @@ The following operators don't fit into any of the other categories:
 
 * ``~``: Converts all operands into strings and concatenates them. ``{{ "Hello
   " ~ name ~ "!" }}`` would return (assuming ``name`` is ``'John'``) ``Hello
-  John!``.
+  John!``. Note that the output will be escaped due to autoescaping.
+  
+  If you wish to have raw output, group the concatenation in parentheses before
+  passing to the raw filter: ``{{ ("Hello " ~ name ~ "!")|raw }}``
```
Comment (xabbuh) [inline doc/templates.rst]: [...] passing them to [...]

File: doc/templates.rst [hunk 3]
```diff
@@ -784,7 +784,10 @@ The following operators don't fit into any of the other categories:
 
 * ``~``: Converts all operands into strings and concatenates them. ``{{ "Hello
   " ~ name ~ "!" }}`` would return (assuming ``name`` is ``'John'``) ``Hello
-  John!``.
+  John!``. Note that the output will be escaped due to autoescaping.
+  
+  If you wish to have raw output, group the concatenation operands in parentheses 
+  before passing them to the raw filter: ``{{ ("Hello " ~ name ~ "!")|raw }}``
```
Comment (fabpot) [inline doc/templates.rst:790]: I don't think that this is the right place. It's more about operator precedence, where we already have the information. I've tried to make it clearer in #2802

**Pull request closed without merging**